modules/libraries/src/backend/utils.ts

```diff
@@ -0,0 +1,178 @@
+import * as sdk from 'botpress/sdk'
+import { spawn } from 'child_process'
+import fse from 'fs-extra'
+import path from 'path'
+
+import { isOffline, packageJsonPath, packageLockJsonPath, sharedLibsDir } from '.'
+
+const debug = DEBUG('libraries')
+
+const LIB_FOLDER = 'libraries/'
+
+const packageNameRegex = /^(@[a-z0-9-~][a-z0-9-._~]*\/)?[a-z0-9-~][a-z0-9-._~]*$/
+
+// Taken from https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+const versionRegex = /^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/
+
+const sanitizeArg = (text: string) => text.replace(/[^a-zA-Z0-9\/_.@^\-\(\) ]/g, '').replace(/\/\//, '/')
+
+export const validateNameVersion = ({ name, version }: { name: string; version?: string }) => {
```

Comment:
I see this function as a good unit test candidate ;) as I'm sure there are many corner cases in name versionning